src/games/supported/DoubleDunk.cpp

```diff
@@ -129,37 +129,79 @@ ModeVect DoubleDunkSettings::getAvailableModes() {
     return modes;
 }
 
+void DoubleDunkSettings::goDown(System &system,
+                            std::unique_ptr<StellaEnvironmentWrapper> &environment) {
+    // this game has a menu that allows to define various yes/no options
+    // this function goes to the next option in the menu
+    unsigned previousSelection = readRam(&system, 0xB0);
+    while(previousSelection == readRam(&system, 0xB0)){
+        environment->act(PLAYER_A_DOWN, PLAYER_B_NOOP);
+        environment->act(PLAYER_A_NOOP, PLAYER_B_NOOP);
+    }
+}
+
+void DoubleDunkSettings::activateOption(System &system, unsigned byteOfInterest,
+                                    std::unique_ptr<StellaEnvironmentWrapper> &environment) {
+    // once we are at the proper option in the menu,
+    // if we want to enable it all we have to do is to go right
+    while((readRam(&system, 0x80) & byteOfInterest) != byteOfInterest) {
+        environment->act(PLAYER_A_RIGHT, PLAYER_B_NOOP);
+        environment->act(PLAYER_A_NOOP, PLAYER_B_NOOP);
+    }
+}
+
+void DoubleDunkSettings::deactivateOption(System &system, unsigned byteOfInterest,
+                                    std::unique_ptr<StellaEnvironmentWrapper> &environment) {
+    // once we are at the proper optio in the menu,
+    // if we want to disable it all we have to do is to go left
+    while((readRam(&system, 0x80) & byteOfInterest) == byteOfInterest) {
+        environment->act(PLAYER_A_LEFT, PLAYER_B_NOOP);
+        environment->act(PLAYER_A_NOOP, PLAYER_B_NOOP);
+    }
+}
+
 // set the mode of the game
 // the given mode must be one returned by the previous function
 void DoubleDunkSettings::setMode(game_mode_t m, System &system,
-                              std::unique_ptr<StellaEnvironmentWrapper> environment) {
+                                std::unique_ptr<StellaEnvironmentWrapper> environment) {
 
     if(m < getNumModes()) {
         environment->pressSelect();
+
+        bool threePts = m % 2 == 1 ? true : false;
```

Comment:
Simpler and clearer: m & 1, m & 2, m & 4, m & 8 (these bits are set).

In fact, you can probably do `threeSecs = (bool) m & 4;`